crm_partner_company_group/views/opportunity_view.xml

```diff
@@ -17,4 +17,43 @@
             </filter>
         </field>
     </record>
+    <record id="crm_lead_view_form" model="ir.ui.view">
+        <field name="name">crm.lead.form</field>
+        <field name="model">crm.lead</field>
+        <field name="inherit_id" ref="crm.crm_lead_view_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//group[@name='opportunity_partner']/field[@name='partner_id']"
+                position="after"
+            >
+                <field name="company_group_id" invisible="not company_group_id" />
```

Comment:
Why not letting to assign it if not set?

Reply:
The field is readonly and copied from the res.partner to crm.lead.

```
class CrmLead(models.Model):
    _inherit = "crm.lead"

    company_group_id = fields.Many2one(
        related="partner_id.company_group_id", store=True
    )

```

I agree that there a some interfaces, where you can change res.partner informations in the lead interface, but this is often not understood correctly by the users and they think they change something on this one lead, but they are actually change it on the linked contact for all leads.

Reply:
OK, understood. Yes, it's OK this way then.